web/client/epics/widgets.js

```diff
@@ -176,5 +176,19 @@ module.exports = {
                 return Rx.Observable.of(
                     ...(has(newProperties, "layerFilter") && flatLayer ? [updateWidgetLayer(flatLayer)] : [])
                 );
+            }),
+
+    /**
+     * Triggers updates of the layer property of widgets on LAYER_LOAD
+     * @memberof epics.widgets
+     * @param {external:Observable} action$ manages `LAYER_LOAD`
+     * @return {external:Observable}
+     */
+    updateLayerOnLoad: (action$, store) =>
+        action$.ofType(LAYER_LOAD)
```

Comment:
Layer load is called many times (one for each tile loaded, if I remember well). The only update needed is when error changed. Is there a way to avoid unnecessary actions flood ? 
I think a distinctUntilKeyChanged on error field could to the job. Please add this improvement and test.

Reply:
Sounds like a good idea.

Reply:
Actually the update is needed when error property is changed for a particular layerId, so distinctUntilKeyChanged would have to be both on layerId and error fields. Would that be particularly useful?

Reply:
note, i updated the example with take(1) after the first edit

Reply:
Edited again. What I'm trying to do is to create a stream for each layerid that errored waiting for load (maybe other checks) to trigger once the updateLayertWidget. Did you get my point?

Anyway this is an alternative solution. The problem is to avoid unnecessary widget's update, even if they don't cause, as far as I can see, reloads or other kind of problems.